src/tests/bulkConcepts/containers/BulkConceptsPage.test.js

```diff
@@ -333,6 +334,38 @@ describe('Test suite for BulkConceptsPage component', () => {
     expect(instance.componentDidUpdate(2)).toEqual(undefined);
   });
 
+  it('it should call the handle componentDidUpdate function while filtering', () => {
+    const props = {
+      setCurrentPage: jest.fn(),
+      currentPage: 1,
+      fetchBulkConcepts: jest.fn(),
+      filterConcept: jest.fn(),
+      concepts: [],
+      loading: true,
+      datatypes: [],
+      classes: [],
+      conceptLimit: 10,
+      match: {
+        params: {
+          type: 'users',
+          typeName: 'emasys',
+          collectionName: 'dev-org',
+          language: 'en',
+          dictionaryName: INTERNAL_MAPPING_DEFAULT_SOURCE,
+        },
+      },
+      addToFilterList: jest.fn(),
+      fetchFilteredConcepts: jest.fn(),
+      addConcept: jest.fn(),
+      previewConcept: jest.fn(),
+      handleNextPage: jest.fn(),
+      componentDidUpdate: jest.fn(),
+    };
+    const wrapper = shallow(<BulkConceptsPage {...props} />).setState({ filterOn: true });
+    const instance = wrapper.instance();
+    expect(instance.componentDidUpdate(2)).toEqual(undefined);
```

Comment:
How is this test related to the pagination fix?

Reply:
@dkayiwa , yes, it's related.

Reply:
How?

Reply:
The test is similar to the one just above it for a normal search and then pagination only that for this particular one, filtering in true in the state.

Reply:
So does this test fail when i set filtering to false?

Reply:
No, it doesn't. Let me modify the test or take it out.

Reply:
Then it is not testing what it should.

Reply:
@dkayiwa I think I made a mistake on using the test just before this one as a template. I have now improved this particular test.